pkg/daemon/ceph/osd/volume.go

```diff
@@ -1156,12 +1156,22 @@ func GetCephVolumeRawOSDs(context *clusterd.Context, clusterInfo *client.Cluster
 			Encrypted:     strings.Contains(blockPath, "-dmcrypt"),
 		}
 
+		diskInfo, err := clusterd.PopulateDeviceInfo(blockPath, context.Executor)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get device info for %q", blockPath)
+		}
+		deviceType := sys.GetDiskDeviceType(diskInfo)
+		osd.DeviceType = deviceType
+		logger.Infof("setting device type %q for device %q", osd.DeviceType, diskInfo.Name)
+
 		if !skipDeviceClass {
```

Comment:
We may want to keep this code for detecting the device type inside the `skipDeviceClass` check. For example, the OSD cleanup case does not need to check for the device type. 
@parth-gr Did you find that the device type wasn't being populated without moving it outside this condition?
@sp98 @satoru-takeuchi wdyt?

Reply:
I can move that code, actually I thought `skipDeviceClass` should just skip device class, but I can move the deviceType code inside it too